framework/src/userobjects/TimedElementSubdomainModifier.C

```diff
@@ -0,0 +1,83 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+// Base class to move elements to a specific subdomain at the given times.
+
+#include "TimedElementSubdomainModifier.h"
+
+registerMooseObject("MooseApp", TimedElementSubdomainModifier);
+
+InputParameters
+TimedElementSubdomainModifier::validParams()
+{
+  InputParameters params = ElementSubdomainModifier::validParams();
+  params.addParam<std::vector<Real>>("times", "The times of the subdomain modifications.");
+
+  return params;
+}
+
+TimedElementSubdomainModifier::TimedElementSubdomainModifier(const InputParameters & parameters)
+  : ElementSubdomainModifier(parameters)
+{
+}
+
+void
+TimedElementSubdomainModifier::initialize()
+{
+  // state variables
+  _last_t = -1;
+  _current_step = _t_step - 1;
+
+  // ask for all times (must NOT be sorted)
+  const auto _times = onGetTimes();
+  const auto n = _times.size();
+
+  // copy data to local storage
+  _times_and_indices.resize(n);
+  for (size_t i = 0; i < n; ++i)
+  {
+    _times_and_indices[i].time = _times[i];
+    _times_and_indices[i].index = i;
+  };
+
+  // sort the times
+  std::sort(_times_and_indices.begin(), _times_and_indices.end());
+}
+
+std::vector<Real>
+TimedElementSubdomainModifier::onGetTimes()
+{
+  std::vector<Real> _times;
+  return _times;
+}
```

Comment:
What's the purpose of this implementation?

Reply:
Thanks for pointing this out. Some code was missing here. Th purpose of "onGetTimes" is to allow inheriting objects to get the (sorted) vector of points in time where changes are to be done.